graphblas_algorithms/algorithms/shortest_paths/dense.py

```diff
@@ -1,14 +1,33 @@
+import numpy as np
 from graphblas import Matrix, Vector, binary, indexunary, replace, select
 from graphblas.semiring import any_plus, any_second
 
-__all__ = ["floyd_warshall", "floyd_warshall_predecessor_and_distance"]
+from ..exceptions import GraphBlasAlgorithmException
+
+__all__ = ["floyd_warshall", "floyd_warshall_predecessor_and_distance", "floyd_warshall_numpy"]
 
 
 def floyd_warshall(G, is_weighted=False):
     return floyd_warshall_predecessor_and_distance(G, is_weighted, compute_predecessors=False)[1]
 
 
-def floyd_warshall_predecessor_and_distance(G, is_weighted=False, *, compute_predecessors=True):
+def floyd_warshall_numpy(G, nodelist=None, is_weighted=False):
```

Comment:
Should we instead name this function `floyd_warshall_matrix` since it doesn't return a numpy array?

Reply:
I actually wonder if this algorithm should exist in our `core`? We could put the `nodelist<->permutation` logic inside nxapi.shortest_paths.dense.floyd_warshall_numpy and have that code call algorithms.shortest_paths.dense.floyd_warshall_predecessor_and_distance directly.

Reply:
Good idea. Done.